commands/pipelines/setup.js

```diff
@@ -0,0 +1,231 @@
+  }, {
+    type: 'confirm',
+    name: 'pull_requests.enabled',
+    message: 'Enable review apps?'
+  }, {
+    type: 'confirm',
+    name: 'pull_requests.auto_deploy',
+    message: 'Automatically create review apps for every PR?',
+    when (answers) { return answers.pull_requests.enabled }
+  }, {
+    type: 'confirm',
+    name: 'pull_requests.auto_destroy',
+    message: 'Automatically destroy idle review apps after 5 days?',
+    when (answers) { return answers.pull_requests.enabled }
+  }])
+}
+
+function* hasCIFlag (heroku) {
+  let hasFlag
+  try {
+    hasFlag = (yield api.getAccountFeature(heroku, 'ci')).enabled
+  } catch (error) {
+    hasFlag = false
+  }
+  return hasFlag
+}
+
+function* getCISettings (organization) {
+  const settings = yield prompt([{
+    type: 'confirm',
+    name: 'ci',
+    message: 'Enable automatic Heroku CI test runs?'
+  }])
+
+  if (settings.ci && organization) {
+    settings.organization = organization
+  }
+
+  return settings
+}
+
+function setupPipeline (token, app, settings, pipelineID, ciSettings = {}) {
+  const promises = [kolkrabbi.updateAppLink(token, app, settings)]
+
+  if (ciSettings.ci) {
+    promises.push(
+      kolkrabbi.updatePipelineRepository(token, pipelineID, ciSettings)
+    )
+  }
+
+  return Promise.all(promises).then(([appLink]) => {
+    return appLink
+  }, (error) => {
+    cli.error(error.response.body.message)
+  })
+}
+
+module.exports = {
+  topic: 'pipelines',
+  command: 'setup',
+  description: 'bootstrap a new pipeline with common settings',
+  help: `Example:
+
+  heroku pipelines:setup example githuborg/reponame -o example-org
+  ? Automatically deploy the master branch to staging? Yes
+  ? Wait for CI to pass before deploying the master branch to staging? Yes
+  ? Enable review apps? Yes
+  ? Automatically create review apps for every PR? Yes
+  ? Automatically destroy idle review apps after 5 days? Yes
+  Creating pipeline... done
+  Linking to repo... done
+  Creating ⬢ example (production app)... done
+  Creating ⬢ example-staging (staging app)... done
+  Configuring pipeline... done
+  View your new pipeline by running \`heroku pipelines:open e5a55ffa-de3f-11e6-a245-3c15c2e6bc1e\``,
+  needsApp: false,
+  needsAuth: true,
+  args: [
+    {
+      name: 'name',
+      description: 'name of pipeline',
+      optional: true
+    },
+    {
+      name: 'repo',
+      description: 'a GitHub repository to connect the pipeline to',
+      optional: true
+    }
+  ],
+  flags: [
+    {
+      name: 'organization',
+      char: 'o',
+      description: 'the organization which will own the apps (aliased as --team)',
+      hasValue: true
+    },
+    {
+      name: 'team',
+      char: 't',
+      description: 'the team which will own the apps (aliased as --organization)',
+      hasValue: true
+    }
+  ],
+  run: cli.command(co.wrap(function*(context, heroku) {
+    const herokuToken = heroku.options.token
+    const githubToken = yield getGitHubToken(herokuToken)
```

Comment:
Rather than passing `herokuToken` and `token` around everywhere, can't we mimic or clone the `heroku` object, and override the properties we need for making calls to Kolkrabbi? If this is a massive hassle, then no worries.